planning/planners/src/encode.rs

```diff
@@ -438,8 +437,10 @@ pub fn add_metric(pb: &FiniteProblem, model: &mut Model, metric: Metric) -> IAto
             let action_costs: Vec<LinearTerm> = costs
                 .iter()
                 .map(|&(ch_id, p, cost)| {
+                    let bounds = model.domain_of(*cost);
+                    assert!(bounds.0 >= 0, "A chronicle could have a negative cost");
                     model
-                        .new_optional_ivar(cost, cost, p, Container::Instance(ch_id).var(VarType::Cost))
+                        .new_optional_ivar(bounds.0, bounds.1, p, Container::Instance(ch_id).var(VarType::Cost))
```

Comment:
Incorrect. The cost you use in the linear term is a new variable that is not linked to actual cost variable (just required to be within its *initial* bounds). 

Why not use the cost variable directly ?

Reply:
Fixed in 031365a72a51bf8c03511dff2126cad5a7c5b9fc.